include/pool/pool_disjoint.h

```diff
@@ -0,0 +1,39 @@
+// Copyright (C) 2023 Intel Corporation
+// Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <umf/memory_pool.h>
+#include <umf/memory_provider.h>
+
+struct umf_disjoint_pool_params {
```

Comment:
We're missing `SharedLimits` here (ref: https://github.com/oneapi-src/unified-memory-framework/pull/4/files#diff-5a84664376e1495e272ca9c9c061388acc9c68940fb5aa9b66e8d9a0c84cc061R36). I'm not sure how to best expose it in C though. The idea behind `SharedLimits` is that it is shared between multiple instances of DisjointPool (.e.g for different types of USM).

We don't have to add this in this PR but let's keep in mind.